compiler/rustc_ast/src/ast.rs

```diff
@@ -585,7 +585,7 @@ impl Pat {
             }
             // A slice/array pattern `[P]` can be reparsed as `[T]`, an unsized array,
             // when `P` can be reparsed as a type `T`.
-            PatKind::Slice(pats) if pats.len() == 1 => pats[0].to_ty().map(TyKind::Slice)?,
+            PatKind::Slice(pats) if let [pat] = &**pats => pat.to_ty().map(TyKind::Slice)?,
```

Comment:
I would kinda prefer we spell this as `pats.as_slice()` or `&pats[..]`.

`&**` is kinda an eyesore, and the implicit autoderef of either strategies makes it less annoying if we change around the underlying type reprs.